Utils/DirectoryWatcher/DirectoryWatcher.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <sys/inotify.h>
+#include <fstream>
+#include <filesystem>
+
+#include <Directory/Directory.h>
+
+#define MAX_EVENTS 128
+#define LEN_NAME 64
+#define EVENT_SIZE ( sizeof (struct inotify_event) )
+#define BUF_LEN ( MAX_EVENTS * ( EVENT_SIZE + LEN_NAME ))
+
+class DirectoryWatcher {
+    int m_fd;
+    std::ofstream log;
+
+public:
+    explicit DirectoryWatcher(const std::filesystem::path& path, const std::filesystem::path& log_path) {
```

Comment:
Consider using `namespace fs = std::filesystem` to shorten `std::filesystem::path`